src/content/reference/cluster-upgrades/index.md

```diff
@@ -0,0 +1,187 @@
+
+- *Major*: A major upgrade occurs when (A) a minor Kubernetes version upgrade is picked up (e. g. from 1.10.* to 1.11.*), when we remove functionality or change functionality that might require changes in workloads, in automation working with the cluster or in administrator's interactions.
+
+**Note:** The release version number is provider-specific.
+Azure, AWS, and KVM based installations have independent versioning systems, as their stacks are also slightly different.
+
+### Determining the release version and inspecting release details
+
+As a user with access to the Giant Swarm API and being member of the organization owning the cluster, you can use both the web UI or the CLI to find out what release version your tenant cluster has currently.
+
+In the web UI, both the cluster overview and the cluster details page show the release version number. 
+In the cluster details page you can click the release version number to get more information about a release.
+
+Likewise in the CLI, commands like [`gsctl list clusters`](/reference/gsctl/list-clusters/) and [`gsctl show cluster`](/reference/gsctl/show-cluster/) reveal the release version number.
+To get information on all available releases, use the [`gsctl list releases`](/reference/gsctl/list-releases/) command.
+
+### Release lifecycle
+
+**We hold two different major releases** available at any time to chose from when creating a new cluster, which means that you have two different Kubernetes minor versions to chose from.
+
+Whenever a new Kubernetes minor version is released by the Kubernetes project, we aim to make that version available in a new major release of our stack within 30 days from the Kubernetes release.
+
+Once we publish a new major release, we deprecate the oldest major release.
+This means that no new clusters can be created using that old release version.
+Existing clusters, however, are not affected.
+
+**Both patch and minor upgrades** can be rolled out at any time by Giant Swarm without your interaction. Currently, this happens in coordination with your administrators and with a notice to your developers.
+
+When a **new major release** becomes available, we inform you, but leave scheduling of the upgrade to you. This gives you the control to decide if and when it is time for you to upgrade, potentially updating workloads first. These upgrades are also acommpanied or even triggered by Giant Swarm staff, to ensure we have a close eye on the upgrade process and the uptime of your workloads.
+
+## How upgrades work
+
+All levels of upgrades, patch, minor, or major, are happening at runtime.
+Your Kubernetes workloads should continue to work as expected (given they meet a few requirements, as explained further below).
+
+In particular this means:
+
+- The master node will be taken down, resulting in the Kubernetes API being shortly unavailable.
+- The worker nodes will be drained and taken down, resulting in pods being rescheduled to other nodes.
+
```

Comment:
I will be more prolific and explain how the mechanism works.

```
The worker nodes are cordon and drain by batches depending on the size of the cluster. Once the new master node is created the first batch of worker nodes spin up. When the upgraded fresh worker nodes are ready, the draining mechanism start and the old instances are terminated. Between every batch, it leaves a fifteen minutes span to ensure all customer workloads have time to start in new nodes. Once the loop is completed, a new batch of nodes is cordoned and drained once the new nodes are ready. 
```
Please can you provide feedback @teemow @xh3b4sd

Reply:
That is provider specific at the moment. Only AWS works in batches. Azure and KVM do one by one. At least this is the desired current implementation. We work on certain issues where this is not true right now but anyway.

Reply:
Also on AWS new instances are created before the old ones get drained and terminated.